Functions/xASL_im_M0ErodeSmoothExtrapolate.m

```diff
@@ -1,18 +1,25 @@
-function [ImOut] = xASL_im_M0ErodeSmoothExtrapolate(ImIn, x)
-
+function [ImOut, VisualQC] = xASL_im_M0ErodeSmoothExtrapolate(ImIn, DirOutput, NameOutput, pvGM, pvWM, brainMap, Quality, LowThreshold)
 %xASL_im_M0ErodeSmoothExtrapolate M0 image processing
 %
-% FORMAT: [ImOut] = xASL_im_M0ErodeSmoothExtrapolate(ImIn, x)
+% FORMAT: [ImOut, VisualQC] = xASL_im_M0ErodeSmoothExtrapolate(ImIn, DirOutput, NameOutput, pvGM, pvWM, brainMap[, Quality, LowThreshold])
 %
 % INPUT:
-%   ImIn - unprocessed M0 image (REQUIRED, 3D image or path)
-%   x    - structure containing fields with all information required to run this submodule (REQUIRED)
+%   ImIn - unprocessed M0 image (3D image or path, REQUIRED)
+%   DirOutput - sring path to output folder, used to be x.D.M0regASLdir (REQUIRED)
+%   NameOutput - string for filename in ['M0_im_proc_' NameOutput '.jpg'], used to be x.P.SubjectID (REQUIRED)
+%   pvGM  - unprocessed pvGM image (3D image or path, same space as M0 image, REQUIRED, used to be x.P.Pop_Path_rc1T1)
+%   pvWM  - unprocessed pvWM image (3D image or path, same space as M0 image, REQUIRED, used to be x.P.Pop_Path_rc2T1)

```

Comment:
You say REQUIRED here, but later on, you have an option for pvGM missing. You need to decide which of the two options you want.